store/iscn.ts

```diff
@@ -86,57 +86,74 @@ export default class ISCN extends VuexModule {
-      }
-    } while (nextSequence !== 0);
-    this.context.commit('setIsLoading', false);
-    return addIDToRecords(records);
+    try {
+      const client = await getQueryClient()
+      let res
+      let nextSequence
+      do {
+        this.context.commit('setIsLoading', true)
+        // eslint-disable-next-line no-await-in-loop
+        res = await client
+          .queryRecordsByOwner(address, nextSequence)
+          .catch(() => {})
+        if (res) {
+          nextSequence = res.nextSequence.toNumber()
+          this.context.commit('appendRecords', res.records)
+          records = records.concat(res.records)
+        }
+      } while (nextSequence !== 0)
+    } catch (error) {
+      console.error(error)
+    } finally {
+      this.context.commit('setIsLoading', false)
+    }
+    return addIDToRecords(records)
   }
 
   @Action
   async queryISCNByKeyword(keyword: string): Promise<ISCNRecordWithID[]> {
     this.context.commit('clearRecords');
-    this.context.commit('setIsLoading', true);
-    const client = await getQueryClient();
-    const [
-      txRes,
-      idRes,
-      fingerprintRes,
-      ownerRes,
-    ] = await Promise.all([
-      client.queryISCNIdsByTx(keyword).catch(() => {}),
-      client.queryRecordsById(keyword).catch(() => {}),
-      client.queryRecordsByFingerprint(keyword).catch(() => {}),
-      client.queryRecordsByOwner(keyword).catch(() => {}),
-    ]);
     let txRecords: ISCNRecord[] = [];
-    if (txRes) {
-      txRecords = (await Promise.all(txRes.map(async (t) => {
-        if (typeof t ==='string') {
-          const res = await client.queryRecordsById(t);
-          return res?.records[0]
-        }
-        return t;
-      }))).filter(t => t) as ISCNRecord[];
+    let result: ISCNRecord[] = ([] as ISCNRecord[])
+    try {
+      this.context.commit('setIsLoading', true)
+      const client = await getQueryClient()
+      const [
+        txRes,
+        idRes,
+        fingerprintRes,
+        ownerRes,
+      ] = await Promise.all([
+        client.queryISCNIdsByTx(keyword).catch(() => {}),
+        client.queryRecordsById(keyword).catch(() => {}),
+        client.queryRecordsByFingerprint(keyword).catch(() => {}),
+        client.queryRecordsByOwner(keyword).catch(() => {}),
+      ])
+      if (txRes) {
+        txRecords = (
+          await Promise.all(
+            txRes.map(async (t) => {
+              if (typeof t === 'string') {
+                const res = await client.queryRecordsById(t)
+                return res?.records[0]
+              }
+              return t
+            }),
+          )
+        ).filter((t) => t) as ISCNRecord[]
+      }
+      result = result
+        .concat(txRecords)
+        .concat(idRes ? idRes.records : [])
+        .concat(fingerprintRes ? fingerprintRes.records : [])
+        .concat(ownerRes ? ownerRes.records : [])
+      this.context.commit('appendRecords', result)
+    } catch (error) {
+      console.error(error)
```

Comment:
Similar to above, is catching the error here intended?

Reply:
I think each request in [Promise.all] should have its own catch, since there might be an error if the keyword format is not suitable for some APIs, I keep another catch outside to handle errors other than above.